### PR TITLE
#166272357 Implement Comment Like Functionality

### PR DESCRIPTION
**Pull request by GerrardE**
#### What does this PR do?
Implements user comment like/unlike functionality

#### Description of Task to be completed?
- Write tests
- Define api endpoints, methods and controllers
- Document routes

#### How should this be manually tested?
1. clone this repo
2. run "npm install"
3. run "npm start"
4. run "npm run test"

#### Any background context you want to provide?
`ALGORITHM`

Step 1: Start
Step 2: Take a new branch out of the develop
Step 3: Push to the remote repository
Step 4: Checkout of the local branch and pull to develop then checkout to the same branch
Step 5: Create database seeders for users and articles,
Step 6: Write tests for the post, get and delete route,
Step 7: Define the API Create, Read and Delete methods in the CommentController,
Step 8: Define the routes in the API routes folder
Step 9: Run tests and ensure they pass
Step 10: Push to the remote repository
Step 11: Stop.

#### What are the relevant pivotal tracker stories?
[#166272357](https://www.pivotaltracker.com/story/show/166272357)

#### Screenshots (if appropriate)
Screenshot 2019-06-20 at 6 26 16 PM

#### Questions:
N/A

### CODE REVIEW

File: server/controllers/comments.js [hunk 1]
```diff
@@ -151,6 +151,93 @@ class CommentController {
...
+   * Get comment like count
+   * @static
+   * @param {object} req - The Request Object
+   * @param {object} res - The Response Object
+   * @param {object} next - The Next Middleware
+   * @return {json} - Returns json Object
+   * @memberof CommentController
+   * @static
+   */
+  static async likeCount(req, res, next) {
```
Comment (fidelisojeah) [inline server/controllers/comments.js]: I really don't see this need for this.

Modify the `singleComment` and `comments` method to include the number of likes as an attribute in the `Comment.findOne` and `Comment.findAll` methods

File: server/routes/api/articles.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ articlesRouter.get('/:slug/comments', Token.authorize, articleFinder, CommentCon
 articlesRouter.put('/:slug/comments/:id', Token.authorize, trim, commentFinder, CommentController.updateComment);
 articlesRouter.delete('/:slug/comments/:id', Token.authorize, commentFinder, CommentController.deleteComment);
 
+articlesRouter.get('/:slug/comments/:id/like', Token.authorize, find, CommentController.likeCount);
+articlesRouter.post('/:slug/comments/:id/like', Token.authorize, find, CommentController.likeComment);
+articlesRouter.delete('/:slug/comments/:id/unlike', Token.authorize, find, CommentController.unlikeComment);
```
Comment (fidelisojeah) [inline server/routes/api/articles.js]:
```suggestion
articlesRouter.delete('/:slug/comments/:id/like', Token.authorize, find, CommentController.unlikeComment);
```

**Review by fidelisojeah — DISMISSED** [review]

File: server/controllers/comments.js (hunk 1 above)
Comment (GerrardE) [reply server/controllers/comments.js]: Feedback resolved.

File: server/routes/api/articles.js (hunk 2 above)
Comment (GerrardE) [reply server/routes/api/articles.js]: Fixed

File: server/helpers/comments.js [hunk 3]
```diff
@@ -18,15 +20,20 @@ export const comments = async (value) => {
...
+    group: ['Comment.id', 'author.id', 'author->profile.id']
   });
 
   return payload;
 };
 
 export const singleComment = async (id) => {
   const payload = await Comment.findOne({
-    attributes: ['id', 'createdAt', 'updatedAt', 'body'],
+    attributes: ['id', 'createdAt', 'updatedAt', 'body', [Sequelize.fn('COUNT', Sequelize.col('CommentLikes.userId')), 'likeCount']],
```
Comment (fidelisojeah) [inline server/helpers/comments.js]: good job.

But if you notice, `Comment.findOne({` and `Comment.findAll({` have similar content (only difference is likely the `where` clause.

I would like to see you employ DRY principles here.

**Review by fidelisojeah — COMMENTED** [review]
Good job so far.

File: server/helpers/comments.js (hunk 3 above)
Comment (GerrardE) [reply server/helpers/comments.js]: I just implemented the feedback. Thanks

**Review by EMacco — APPROVED** [review]

**Review by despeauxz — APPROVED** [review]
Good job

**Pull request merged**